app/lib/submission_builder/return_w2.rb

```diff
@@ -2,47 +2,58 @@ module SubmissionBuilder
         if state_tax_group_xml.present?
           state_local_tax_grp_node.inner_html = state_tax_group_xml
         else
           state_local_tax_grp_node.remove
         end
 
-        existing_stpickup = xml_node.at_xpath("//OtherDeductionsBenefitsGrp[Desc='STPICKUP']")
-
-        if intake_w2.box14_stpickup.present? && intake_w2.box14_stpickup.positive?
-          if existing_stpickup
-            existing_stpickup.at('Amt').content = intake_w2.box14_stpickup.round.to_s
-          else
-            stpickup_node = Nokogiri::XML::Node.new('OtherDeductionsBenefitsGrp', xml_node)
-
-            desc_node = Nokogiri::XML::Node.new('Desc', xml_node)
-            desc_node.content = 'STPICKUP'
-            amt_node = Nokogiri::XML::Node.new('Amt', xml_node)
-            amt_node.content = intake_w2.box14_stpickup.round.to_s
-
-            stpickup_node.add_child(desc_node)
-            stpickup_node.add_child(amt_node)
-
-            if state_local_tax_grp_node.present?
-              state_local_tax_grp_node.add_previous_sibling(stpickup_node)
-            else
-              xml_node.at(:StandardOrNonStandardCd).add_previous_sibling(stpickup_node)
-            end
-          end
+        node_after_box_14_codes = state_local_tax_grp_node
+        unless state_local_tax_grp_node.present?
+          node_after_box_14_codes = @xml_node.at(:StandardOrNonStandardCd)
+        end
+        state_code = @intake_w2.state_file_intake.state_code
+        box14_codes = StateFile::StateInformationService.w2_supported_box14_codes(state_code)
+        box14_codes.each do |code|
+          add_box_14_node(code, node_after_box_14_codes)
         end
       end
         
-      locality_nm = xml_node.at(:LocalityNm)
+      locality_nm = @xml_node.at(:LocalityNm)
       if locality_nm.present?
         locality_nm.inner_html = locality_nm.inner_html.upcase
       end
-      xml_node
+      @xml_node
+    end
+
+    def add_box_14_node(code, node_after_box_14_codes)
+      field_name = "box14_#{code.downcase}".to_sym
+      field_as_desc = code.tr '_', '/' 
+      existing_xml_node = @xml_node.at_xpath("//OtherDeductionsBenefitsGrp[Desc='#{field_as_desc}']")
+
+      if @intake_w2[field_name].present? && @intake_w2[field_name].positive?
+        if existing_xml_node
+          existing_xml_node.at('Amt').content = @intake_w2[field_name].round.to_s
+        else
+          this_code_node = Nokogiri::XML::Node.new('OtherDeductionsBenefitsGrp', @xml_node)
+
+          desc_node = Nokogiri::XML::Node.new('Desc', @xml_node)
+          desc_node.content = field_as_desc
+          amt_node = Nokogiri::XML::Node.new('Amt', @xml_node)
+          amt_node.content = @intake_w2[field_name].round.to_s
+
+          this_code_node.add_child(desc_node)
+          this_code_node.add_child(amt_node)
+
+          node_after_box_14_codes.add_previous_sibling(this_code_node)
+        end
+      end
```

Comment:
[not sure how to categorize this] - I'm surprised/wondering why there isn't a test file in `spec/lib/submission_builder/return_w2`, that could have test expectations to confirm the new code here is working correctly. Is that something that should be in the scope of this PR (which would make this a boulder) or am I missing a reason why that's not needed (which would make this dust)?

Reply:
i think this is actually tested in `state_return_spec.rb` (it should probably be moved into a file called `return_w2_spec.rb` but that hasn't happened yet). i _think_ that should make sure this is still working correctly...